File: src/main/java/io/github/komdosh/alarms/actions/Alarm20Min.java
package io.github.komdosh.alarms.actions;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import io.github.komdosh.alarms.Alarms;

import java.util.concurrent.TimeUnit;

public class Alarm20Min extends AnAction {

    public Alarm20Min() {
        super("Alarm20Min");
    }

    public void actionPerformed(AnActionEvent event) {
        Project p = event.getProject();

        Alarms.setAlertInSeconds(p, TimeUnit.MINUTES.toSeconds(20), "Minutes");
    }
}

File: src/main/kotlin/io/github/komdosh/alarms/actions/Alarm20Min.kt
package io.github.komdosh.alarms.actions

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.ServiceManager
import io.github.komdosh.alarms.services.AlarmService
import java.util.concurrent.TimeUnit

class Alarm20Min : AnAction("Alarm20Min") {
    override fun actionPerformed(event: AnActionEvent) {
        val p = event.project

        val alarmService = ServiceManager.getService(AlarmService::class.java)
        alarmService.setAlertInSeconds(p!!, TimeUnit.MINUTES.toSeconds(20), "minutes")
    }
}